daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala

```diff
@@ -320,6 +322,10 @@ private[lf] case class PartialTransaction(
       sb.toString
     }
 
+  private def normalizedNodeSeeds(): NodeSeeds = {
+    ImmArray(actionNodeSeeds.toImmArray.toList.zipWithIndex.map { case (v, i) => (NodeId(i), v) })
```

Comment:
You should also avoid using `ImmArray#apply[X](xs: X*)` which is here more for convenience that for efficiency.

The optimal version going in the direction of @cocreature is as follow:
```suggestion
    actionNodeSeeds.toImmArray.toSeq.zipWithIndex.map { case (v, i) => (NodeId(i), v) }.toImmArray
```